nntrainer/compiler/tflite_interpreter.cpp

```diff
@@ -56,6 +60,27 @@ void builder2file(const flatbuffers::FlatBufferBuilder &builder,
   os.close();
 }
 
+/**
+ * @brief get predecessor nodes
+ *
+ * @param node the node from which to get predecessor nodes
+ * @note virtual nodes are ignored
+ */
+std::vector<const TfOpNode *> getPredNodes(const TfOpNode &node) {
+  std::vector<const TfOpNode *> predNodes;
+
+  for (auto input : node.getInputNodes()) {
+    const TfOpNode *pred = input;
+    while (pred->isVirtualNode()) {
+      /// Assume that virtual nodes have single input
+      assert(pred->arity() == 1);
+      pred = pred->arg(0);
+    }
+    predNodes.push_back(pred);
```

Comment:
I don't get it. Why skip the non-virtual nodes? Anyway, the function doesn't care about inputs' index.

Reply:
If there is non-virtual node between nodes, predNode and original node index will not 1:1 map.
If no one care about index, I guess it will be fine :)